### ISSUE TITLE
Scratch control sequences in \define@choicekey

### ISSUE THREAD

**Issue by eg9** [reutenauer/polyglossia#390]
The examples in the documentation of `xkeyval` use `\val\nr` for choice keys, but this is bad programming, because this would overwrite definitions of `\val` and `\nr` in the document preamble.

The control sequences are just scratch ones and should be private. Taking `gloss-welsh.ldf` as example,
```latex
\define@choicekey*+{welsh}{date}[\val\nr]{long,short}[short]{%
   \ifcase\nr\relax
      % long:
      \welsh@formaldatetrue
   \or
      % accented:
      \welsh@formaldatefalse
   \fi
   \xpg@info{Option: Welsh, date=\val}%
}{\xpg@warning{Unknown date value `#1'}}
```
should better be
```latex
\define@choicekey*+{welsh}{date}[\xpg@val\xpg@nr]{long,short}[short]{%
   \ifcase\xpg@nr\relax
      % long:
      \welsh@formaldatetrue
   \or
      % accented:
      \welsh@formaldatefalse
   \fi
   \xpg@info{Option: Welsh, date=\xpg@val}%
}{\xpg@warning{Unknown date value `#1'}}
```
There is nothing magic in the two names; the optional argument should contain two control sequences that are used in the fashion shown in the code above.

Reference: https://tex.stackexchange.com/q/527681/4427

**Comment by jspitz** [comment]
That's indeed not apparent from the xkeyval documentation.

**Comment by jspitz** [comment]
Fixed for 1.48.
